### DISCUSSION TITLE
Sign commits

### DISCUSSION THREAD

**Discussion by andreiblt** [jesseduffield/lazygit#3738] [Q&A] [answered]
Hello there, I just started using this plugin and I'm in love with it, but I also noticed that it isn't automatically signs my commits. Is there something I should do so it signs all the commits I am writing ?

**Comment by stefanhaller** [comment, marked as answer]
I never sign commits and don't have experience with it, so I might be off here, but as far as I understand, this has nothing to do with lazygit. You configure this in git by saying `git config --global commit.gpgsign true`.

**Reply by andreiblt** [reply]
Indeed, when I signed commits I manually did that with `git commit -S` in the terminal. The `global git config` worked! Thank you for you answer and for the awesome plugin !

**Reply by MLengl** [reply]
Facing same questions on loveley lazygit because my project enforces signing commits as default.

Signing can work automatically:
```
// General git signing config
git config --global gpg.format ssh
git config --global commit.gpgsign true
git config --global user.signingkey ~/.ssh/id_ed25519.pub

// Just allow special signing files locally
mkdir -p ~/.config/git
touch ~/.config/git/allowed_signers
echo max.mustermann@hismail.com $(cat ~/.ssh/id_ed25519.pub)" >> ~/.config/git/allowed_signers 
git config --global gpg.ssh.allowedSignersFile ~/.config/git/allowed_signers
```

With this git will automatically sign every commit that is made. -S option is not needed explicitly.

Anyway now I want to verify and check it commits are signed or not.
Maybe Lazygit could get just a indicator label/badge for commits that are signed. Details can anyway be showed/checked using `git log --show-signature`

Showing insights directly might be a positiv and modern sign in good direction.

**Comment by miko007** [comment]
i usually only sign release commits. it would be nice if i could trigger `commit -S` from within lazygit, is that a possibility?

**Reply by stefanhaller** [reply]
Only using a [custom command](https://github.com/jesseduffield/lazygit/blob/master/docs/Custom_Command_Keybindings.md) right now.

**Reply by miko007** [reply]
that works for me, thank you very much!
